tsl/src/continuous_aggs/create.c

```diff
@@ -514,11 +515,9 @@ mattablecolumninfo_create_materialization_table(MatTableColumnInfo *matcolinfo,
 
 	/* Initialize the invalidation log for the cagg. Initially, everything is
 	 * invalid. */
-	if (OidIsValid(origquery_tblinfo->nowfunc))
-		current_time = ts_time_value_to_internal(OidFunctionCall0(origquery_tblinfo->nowfunc),
-												 get_func_rettype(origquery_tblinfo->nowfunc));
-	else
-		current_time = GetCurrentTransactionStartTimestamp();
+	mat_time_dim = hyperspace_get_open_dimension(ht->space, 0);
+	Assert(mat_time_dim != NULL);
+	current_time = ts_get_now_internal(mat_time_dim);
```

Comment:
It looks like the new code does not call the `integer_now` function for the hypertable. Could you elaborate in the commit message why this is actually not the case?

Also, since the `nowfunc` field is filled in from the dimension, do we still need this field?

Reply:
ts_get_now_internal handles all the different OID cases for the time dimension. So the change is to use the common api rather than make type specific calls.

Reply:
removed nowfunc from the struct.